pkg/injector/bootstrap.go

```diff
@@ -101,3 +105,118 @@ func (wh *mutatingWebhook) marshalAndSaveBootstrap(name, namespace string, confi
+				Msgf("Error getting cert %s from secret %s/%s", bootstrapCert, secret.Namespace, secret.Name)
+		}
+
+		if !b.certManager.ShouldRotate(bootstrapCert) {
+			continue
+		}
+
+		issuedCert, err := b.certManager.IssueCertificate(bootstrapCert.CommonName.String(), certificate.Internal, certificate.FullCNProvided())
+		if err != nil {
+			log.Error().Err(err).Str(errcode.Kind, errcode.GetErrCodeWithMetric(errcode.ErrRotatingCert)).Msgf("Error rotating cert %s", issuedCert)
+		}
+
+		secret.Data[bootstrap.EnvoyXDSCACertFile] = issuedCert.GetTrustedCAs()
+		secret.Data[bootstrap.EnvoyXDSCertFile] = issuedCert.GetCertificateChain()
+		secret.Data[bootstrap.EnvoyXDSKeyFile] = issuedCert.GetPrivateKey()
+
+		err = b.kubeController.UpdateSecretData(ctx, secret, secret.Data)
+		if err != nil {
+			if apierrors.IsConflict(err) {
+				log.Error().Err(err).Str(errcode.Kind, errcode.GetErrCodeWithMetric(errcode.ErrUpdatingBootstrapSecret)).
+					Msgf("There was an update conflict while trying to update the envoy bootstrap config secret %s/%s with issued cert %s", secret.Namespace, secret.Name, issuedCert)
+				continue
+			}
+			log.Error().Err(err).Str(errcode.Kind, errcode.GetErrCodeWithMetric(errcode.ErrUpdatingBootstrapSecret)).
+				Msgf("Error updating bootstrap secret %s/%s with issued cert %s", secret.Namespace, secret.Name, issuedCert)
+		}
+	}
+}
+
+func getCertFromSecret(secret *corev1.Secret) (*certificate.Certificate, error) {
+	pemCert, ok := secret.Data[bootstrap.EnvoyXDSCertFile]
+	if !ok {
+		log.Error().Err(certificate.ErrInvalidCertSecret).Str(errcode.Kind, errcode.GetErrCodeWithMetric(errcode.ErrObtainingCertFromSecret)).
+			Msgf("Opaque k8s secret %s/%s does not have required field %q", secret.Namespace, secret.Name, bootstrap.EnvoyXDSCertFile)
+		return nil, certificate.ErrInvalidCertSecret
+	}
+
+	pemKey, ok := secret.Data[bootstrap.EnvoyXDSKeyFile]
+	if !ok {
+		log.Error().Err(certificate.ErrInvalidCertSecret).Str(errcode.Kind, errcode.GetErrCodeWithMetric(errcode.ErrObtainingPrivateKeyFromSecret)).
+			Msgf("Opaque k8s secret %s/%s does not have required field %q", secret.Namespace, secret.Name, bootstrap.EnvoyXDSKeyFile)
+		return nil, certificate.ErrInvalidCertSecret
+	}
+
+	caCert, ok := secret.Data[bootstrap.EnvoyXDSCACertFile]
+	if !ok {
+		log.Error().Err(certificate.ErrInvalidCertSecret).Str(errcode.Kind, errcode.GetErrCodeWithMetric(errcode.ErrObtainingCACertFromSecret)).
+			Msgf("Opaque k8s secret %s/%s does not have required field %q", secret.Namespace, secret.Name, bootstrap.EnvoyXDSCACertFile)
+		return nil, certificate.ErrInvalidCertSecret
+	}
+
+	x509Cert, err := certificate.DecodePEMCertificate(pemCert)
+	if err != nil {
+		log.Error().Err(err).Str(errcode.Kind, errcode.GetErrCodeWithMetric(errcode.ErrDecodingPEMCert)).
+			Msg("Error converting PEM cert to x509 to obtain serial number")
+		return nil, err
+	}
+
+	return &certificate.Certificate{
```

Comment:
Currently, the `siginingIssuerID` and the `validatingIssuerID` will always be empty strings. When we call `ShouldRotate()` the certificate's `siginingIssuerID` and `validatingIssuerID`  will never match the Managers `signingIssuer.ID` and `validatingIssuer.ID` respectively since they will be set to the name of the appropriate MRC resource. Therefore, `ShouldRotate()` will always return true and the certificates will constantly be rotated. 

To resolve this, we could add additional data fields to the osm-bootrstrap secret for the `siginingIssuerID` and `validatingIssuerID`. `GetCertFromSecret` could access those values to create a `Certificate`. Since the `siginingIssuerID` and `validatingIssuerID` are not exported, we could add a `Certificate` builder method like `NewCertificate()` in pkg/certificate.